sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero_usm.cpp

```diff
@@ -518,28 +528,81 @@ static ur_result_t USMAllocationMakeResident(
   return UR_RESULT_SUCCESS;
 }
 
+ur_usm_pool_handle_t_::ur_usm_pool_handle_t_(ur_context_handle_t Context,
+                                             ur_usm_pool_desc_t *PoolDesc) {
+
+  zeroInit = static_cast<ur_bool_t>(PoolDesc->flags &
+                                    UR_USM_POOL_FLAG_ZERO_INITIALIZE_BLOCK);
+
+  void *pNext = const_cast<void *>(PoolDesc->pNext);
+  while (pNext != nullptr) {
+    const ur_base_desc_t *BaseDesc =
+        reinterpret_cast<const ur_base_desc_t *>(pNext);
+    switch (BaseDesc->stype) {
+    case UR_STRUCTURE_TYPE_USM_POOL_LIMITS_DESC: {
+      const ur_usm_pool_limits_desc_t *Limits =
+          reinterpret_cast<const ur_usm_pool_limits_desc_t *>(BaseDesc);
+      for (auto &config : USMAllocatorConfigs.Configs) {
+        config.MaxPoolableSize = Limits->maxPoolableSize;
+        config.SlabMinSize = Limits->minDriverAllocSize;
+      }
+      break;
+    }
+    default: {
+      urPrint("urUSMPoolCreate: unexpected chained stype\n");
+      throw UsmAllocationException(UR_RESULT_ERROR_INVALID_ARGUMENT);
+    }
+    }
+    pNext = const_cast<void *>(BaseDesc->pNext);
+  }
+
+  HostMemPool = std::make_unique<USMAllocContext>(
+      std::unique_ptr<SystemMemory>(new USMHostMemoryAlloc(Context)),
+      this->USMAllocatorConfigs.Configs[usm_settings::MemType::Host]);
+
+  for (auto device : Context->Devices) {
+    DeviceMemPools[device] = std::make_unique<USMAllocContext>(
+        std::unique_ptr<SystemMemory>(
+            new USMDeviceMemoryAlloc(Context, device)),
```

Comment:
should we have try/catch for these new statements?

Reply:
The whole constructor is called inside a try/catch block that will catch any exceptions new throws